### ISSUE TITLE
$SHARE_DIR cleanup on install???

### ISSUE THREAD

**Issue by tpruzina** [avast-tl/retdec#12]
While not exactly recommended by README, If I ever wanted to install into /usr or /usr/local prefix your ```retdec/cmake/install-share.sh``` would ```rm -rf``` my $SHARE_DIR.

Please fix before somebody tries to install into their /usr.

> tomas@tpruzina build # sandbox.py -p make-install make install
...
-- Install configuration: "Release"
**rm: cannot remove '/usr/local/share/doc/vlc/libvlc/QtPlayer': Read-only file system
...
rm: cannot remove '/usr/local/share/wine': Read-only file system**
RUN: wget https://github.com/avast-tl/retdec-support/releases/download/2017-12-12/retdec-support_2017-12-12.tar.xz -O /usr/local/retdec-support_2017-12-12.tar.xz
/usr/local/retdec-support_2017-12-12.tar.xz: Read-only file system
ERROR: wget failed
**rm: cannot remove '/usr/local/share/doc/vlc/libvlc/QtPlayer': Read-only file system
...
rm: cannot remove '/usr/local/share/doc/vlc/lua/http/requests': Read-only file system
rm: cannot remove '/usr/local/share/wine': Read-only file system**
CMake Error at cmake_install.cmake:47 (message):
  RetDec share directory installation FAILED

**Comment by PeterMatula** [comment]
Well, I must say that after a long time playing with it only in our set environments, we were kinda expecting something like this when released. But rm-rf-ing parts of `/usr` is definitely AN issue.

We will not be able to fix it today, but I at least added a warning in c1206a6, and solving this will be the first task tomorrow.

**Comment by s3rvac** [comment]
Thank you very much for your report! In 03ea29a27fd82986bef28eb3a484a0f8595020ce, I have modified the script to not remove the entire `share` directory. Instead, it will now remove only subdirectories and files that are used by RetDec. This should reduce the chance of unintentionally removing system directories when RetDec is accidentally installed on Linux into e.g. `/usr` as the `arm`, `generic`, and `x86` directories are not commonly used by Linux distributions.

Nevertheless, until our build system properly supports system-wide installation, RetDec should be installed into a clean, dedicated directory.

**Comment by PeterMatula** [comment]
After fb9b340 and #31, installation creates:
- `bin/`
- `lib/libretdec-{libelf,libdwarf}.so` (Linux)
- `share/retdec/doc`
- `share/retdec/support`

Only `share/retdec/support` is rm-rf-ed in the installation step. I looked for more potentially dangerous `rm` commands, but found none, so lets hope there are no more surprises like this.

After this, we are one step closer to a system-wide installation. However, I still would not recommend it -- there are to many binaries, scripts, and stuff in `bin`. Closing this, potential system-wide installation may be solved later in some dedicated issue.
